### DISCUSSION TITLE
Implementation of PinMode RTC_GPIO_MODE_INPUT_OUTPUT_OD

### DISCUSSION THREAD

**Discussion by JiriRotter** [espressif/arduino-esp32#8066] [Q&A]
Hello,

i possible with current arduino-esp32 use pinmode Input+Output Open Drain? ESP32 is capable of it like it is defined (enumerated) in esp-idf. But in arduino-esp32/cores/esp32/esp32-hal-gpio.h this only pinmode is not defined.

Unfortunately I haven't found out where and how are those enumeration used. I have only found out, that 
`pinMode(OUTPUT,(OUTPUT_OPEN_DRAIN | INPUT));`
is not working for me. CPU is unable to read the state of the pin.

**Comment by mrengineer7777** [comment]
Try `pinMode(PIN_NUMBER, (OUTPUT | OPEN_DRAIN));`
Note that "OUTPUT" is a #define.  Don't use it in the first parameter.

**Reply by mrengineer7777** [reply]
See esp32-hal-gpio.c and .h for details.  `pinMode` maps to `__pinMode`.

**Reply by JiriRotter** [reply]
Sorry about the name of the "PIN". I have edited it (not using the name in the program) and haven't noticed, that it can't be compiled.

Iam trying to write my I2C simple stack in need of very low frequency of the bus (max 50kHz) of application. I have pull-up resistor on the bus as required. But when I read back pin I2C_SDA after DigitalWrite to monitor values Iam sending, the result is still 0, no metter if Iam sending 1 or 0. So I wonder, if I have correct pinmode defined and if is even possible in arduino-esp32 to have this mode active if it is not directly enumerated.

Iam trying to workaround it with this pinMode(I2C_SCL,(OUTPUT_OPEN_DRAIN | INPUT));

**Reply by mrengineer7777** [reply]
Are you toggling the pin yourself (Software I2C) or using the I2C library (wire.h)?
